src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -711,6 +711,31 @@ public Response validateCondition(Policy policy) {
 
   }
 
+  @Operation(summary = "Validates a Policy condition")
+  @POST
+  @Path("/validate-name")
+  @APIResponses({
+          @APIResponse(responseCode = "200", description = "Name validated"),
+          @APIResponse(responseCode = "400", description = "Name not valid"),
+          @APIResponse(responseCode = "500", description = "No policy provided or internal error")
+  })
+  public Response validateName(Policy policy) {
```

Comment:
Policy should be `@Valid` and (see my other PR) `@NotNull`. 
The latter also removes the need for the null check a few lines down.

Reply:
I think I'll skip the `@Valid` or i would have to provide a dummy condition for this call (but will use the `@NotNull`.

Any concern with that?

Reply:
I guess it is ok if we/you document this.
And then, if it is only about name validation, we could only pass a String that needs to fit the  rules for a name (not empty, max 150 chars) and not a Policy object.

Reply:
Ah, I suppose I could just pass the name. 

I wanted to keep the name validation, but without `@Valid` is not being used anyway.

Reply:
Ok, changed it to receive the name in the body, and the Id (in case is an existing policy) with query param.